### DISCUSSION TITLE
The QRFpdf output requires FPDF

### DISCUSSION THREAD

**Discussion by sandikodev** [chillerlan/php-qrcode#93] [Q&A]
i use dev branch btw
```
{
    "require": {
        "chillerlan/php-qrcode": "dev-main",
     }
}

```
then iam just use and follow this pieces of code
```
https://github.com/chillerlan/php-qrcode/blob/main/examples/fpdf.php
```
and here what i got
```
[Sat Nov 13 22:39:32.650718 2021] [php7:error] [pid 22991] [client 127.0.0.1:41746] PHP Fatal error:  Uncaught chillerlan\\QRCode\\QRCodeException: The QRFpdf output requires FPDF as dependency but the class "\\FPDF" couldn't be found. in /app/vendor/chillerlan/php-qrcode/src/Output/QRFpdf.php:36\nStack trace:\n#0 /app/vendor/chillerlan/php-qrcode/src/QRCode.php(249): chillerlan\\QRCode\\Output\\QRFpdf->__construct()\n#1 /appvendor/chillerlan/php-qrcode/src/QRCode.php(180): chillerlan\\QRCode\\QRCode->initOutputInterface()\n#2 /app/okek.php(46): chillerlan\\QRCode\\QRCode->render()\n#3 {main}\n  thrown in /app/vendor/chillerlan/php-qrcode/src/Output/QRFpdf.php on line 36
```

so here kind of browser would be looks like
![Screenshot from 2021-11-13 22-49-48](https://user-images.githubusercontent.com/33443311/141650218-881a96f2-0cd0-414c-a808-067cf68f8ca8.png)

May what i need to ask. Why i can't meet `FPDF` automaticaly ?
if So. There some both repos, the first one from `Setasign/FPDF`
that must be default of `chillerlan/php-qrcode` use and ther's another from `fpdf/fpdf`,
which either work well when i add them as my `composer require` module

but for `fpdf/fpdf` i need small little change `use Fpdf\Fpdf` instead `use FPDF` in `/app/vendor/chillerlan/php-qrcode/src/Output/QRFpdf.php`

as `tecnickcom/tcpdf` alternative, this help me alot with more specific customization
Big Thanks

**Comment by codemasher** [comment]
It's an optional library and if you want to use the FPDF output, you'll need to install it by yourself. Just add it to your `composer.json` like so:

```json
{
    "require": {
        "chillerlan/php-qrcode": "^4.3",
	"setasign/fpdf": "^1.8.2"
     }
}
```

**Comment by codemasher** [comment]
Aside, `tecnickcom/tcpdf` is a completely different library and if someone wants support for this, they're welcone to write an output plugin for it.
